commit 6b42a20e70fbba3f24d21ecdc29014337527127d
Author: martin f. krafft <madduck@debian.org>
Date:   Sat Jan 2 12:58:49 2010 +1300

    Ensure target locations exists before writing template files
    
    scripts/wicd-client could not be written since scripts/ didn't exist.
    This patch creates directories as needed, and also attempts to remove
    emmpty directories on clean.
    
    WARNING: it seems that the cleangenerated class is never called though.
    
    Signed-off-by: martin f. krafft <madduck@debian.org>

---
 setup.py |    4 ++++
 1 file changed, 4 insertions(+)

--- wicd.orig/setup.py
+++ wicd/setup.py
@@ -332,6 +332,10 @@ class configure(Command):
                 original_name = os.path.join('in',item)
                 item_in = open(original_name, 'r')
                 final_name = item[:-3].replace('=','/')
+                parent_dir = os.path.dirname(final_name)
+                if len(parent_dir) > 0 and not os.path.exists(parent_dir):
+                    print '(mkdir %s)'%parent_dir,
+                    os.makedirs(parent_dir)
                 print final_name
                 item_out = open(final_name, 'w')
                 for line in item_in.readlines():
